primitives/state-machine/src/basic.rs

```diff
@@ -211,9 +211,9 @@ impl Externalities for BasicExternalities {
 		&mut self,
 		child_info: &ChildInfo,
 		_limit: Option<u32>,
-	) -> bool {
+	) -> (bool, u32) {
 		self.inner.children_default.remove(child_info.storage_key());
-		true
+		(true, 1)
```

Comment:
Is returning `1` right here?

Reply:
This implementation disregards the `limit` and deletes all keys because it can in `O(1)`. Therefore the hardcoded `true`. You would need to find out how many that were. I think that should work:
```rust
let num_removed = self.inner.children_default.remove(child.info.storage_key()).map(|c| c.data.len()).unwrap_or(0);
```